### PR TITLE
Fix e2e tests

### PR DESCRIPTION
**Pull request by jhellar**
## Motivation
https://issues.jboss.org/browse/RAINCATCH-1253

## Verification
- have mongo and redis servers running
- have raincatcher core and angularjs running
- have selenium server running
  - `cd tests/e2e-ui-tests`
  - `npm install`
  - `npm run wdm:update`
  - `npm run wdm:start`
- start the tests
  - `cd tests/e2e-ui-tests`
  - `npm run test-headless:ui -- --suite=portal`
  - `npm run test-headless:ui -- --suite=mobile` (this test will fail because of https://issues.jboss.org/browse/RAINCATCH-1279)

@twnolan @wtrocki  ping

### CODE REVIEW

File: tests/e2e-ui-tests/utils/constants.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+module.exports = {
```
Comment (twnolan) [inline tests/e2e-ui-tests/utils/constants.js]: This entire file is no longer needed. It has been replaced by the files in the data folder which splits the data into various sections.

File: tests/e2e-ui-tests/utils/constants.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    AUTH_FAIL_MSG: 'Authentication Failed! Try Again.',
+    VALUE_ATTRIBUTE: 'value',
+    USERNAME_MISSING_MSG: 'A username is required.',
+    PASSWORD_MISSING_MSG: 'A password is required.'
+  },
+  logout: {
+    URL: '/login',
+    DEFAULT_BODY: 'Trouble logging in? Contact the switchboard.'
+  },
+  schedule: {
```
Comment (twnolan) [inline tests/e2e-ui-tests/utils/constants.js]: schedule, groups, workers, files etc. have all been removed from the applications

File: tests/e2e-ui-tests/utils/constants.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    LONGITUDE_MISSING_MSG: 'An longitude is required.',
+    FINISH_DATE_MISSING_MSG: 'A finish date is required.',
+    FINISH_TIME_MISSING_MSG: 'A finish date is required.',
+    SUMMARY_MISSING_MSG: 'A summary date is required.'
+  },
+  files: {
+    URL: '/files',
+    URL_DETAIL: '/files/detail/'
+  },
+  dateFormat: {
```
Comment (twnolan) [inline tests/e2e-ui-tests/utils/constants.js]: dates days and months and formating are no longer needed

File: tests/e2e-ui-tests/tests/mobile/workorder-workflow.spec.js [hunk 4]
```diff
@@ -1,159 +1,107 @@
...
-      });
+    step('click back button', function() {
+      workflowService.cancelWorkflowChanges();
+    });
 
-      step('verify there is only one workorder in the list', function () {
-        workorderService.verifyNumberOfWorkordersInList(1);
-      });
+    step('verify that the begin workflow button is still present', function() {
+      browser.sleep(1000);
```
Comment (twnolan) [inline tests/e2e-ui-tests/tests/mobile/workorder-workflow.spec.js:85]: We shouldn't use a sleep value in tests. If you are waiting for an element to appear/disappear, then you can use the functions in `utils/wait.js`. 

This should appear inside the `verifyWorkorderWorkflowIsNotCompleted()` function of workorder service as the functions in the service object should contain all information associated with completing the tests.

File: tests/e2e-ui-tests/data/page_constants.js [hunk 5]
```diff
@@ -1,25 +1,28 @@
 module.exports = {
```
Comment (twnolan) [inline tests/e2e-ui-tests/data/page_constants.js:1]: One thing we could eventually do is move these page values into their own page objects, but that is an improvement we can do in the future

**Review by twnolan — APPROVED** [review]
This is great work so far. Well done. The only things that really stand out are:

- utils/constants.js is not needed as we have replaced this with the data folder
- and the use of sleeps in the test files when we have functions in utilities that can be used instead.

utils/constants.js should be removed but other than that this is ok to be merged. The sleeps in test files can be changed out when we are making test improvements in the future

**Review by twnolan — APPROVED** [review]

File: tests/e2e-ui-tests/tests/mobile/workorder-workflow.spec.js (hunk 4 above)
Comment (jhellar) [reply tests/e2e-ui-tests/tests/mobile/workorder-workflow.spec.js:85]: Created a jira for that https://issues.jboss.org/browse/RAINCATCH-1284

**Comment by jhellar** [comment]
@twnolan thanks for review. Constants file is removed.

**Review by twnolan — APPROVED** [review]

**Review by wtrocki — APPROVED** [review]

**Pull request merged**